prow/config/jobs_test.go

```diff
@@ -76,8 +79,12 @@ func TestPresubmits(t *testing.T) {
 			}
 			// Check that the merge bot will run AlwaysRun jobs, otherwise it
 			// will attempt to rerun forever.
-			if job.AlwaysRun && !job.re.MatchString(testThis) {
-				t.Errorf("AlwaysRun job %s: \"%s\" does not match regex \"%v\".", job.Name, testThis, job.Trigger)
+			if job.AlwaysRun && !job.re.MatchString(newTestThis) && !job.re.MatchString(oldTestThis) {
+				t.Errorf("AlwaysRun job %s: \"%s\" and \"%s\" do not match regex \"%v\".", job.Name, newTestThis, oldTestThis, job.Trigger)
+			}
+			// Check that the merge bot will not run Non-AlwaysRun jobs
+			if !job.AlwaysRun && (job.re.MatchString(newTestThis) || job.re.MatchString(oldTestThis)) {
```

Comment:
But we want to validate that **both** newTestThis and oldTestThis matches when AlwaysRun.